h2o-py/tests/testdir_algos/glm/pyunit_PUBDEV_7645_lambda_search_warning.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import print_function
+from builtins import range
+import sys, os
+sys.path.insert(1,"../../../")
+import h2o
+from h2o.estimators.glm import H2OGeneralizedLinearEstimator
+from tests import pyunit_utils
+
+try:    # redirect python output
+    from StringIO import StringIO  # for python 3
+except ImportError:
+    from io import StringIO  # for python 2
+
+# This method test to make sure when a user set lambda_search=True and choose a lambda value,  a warning message 
+# should be generated and passed to the user.
+def test_benign():
+    training_data = h2o.import_file(pyunit_utils.locate("smalldata/gbm_test/BostonHousing.csv"))
+    Y = 13
+    X = list(range(13))
+    check_warnings("lambda_search:  will be disabled if you specify lambda values.", X, Y, 1, training_data)   # expect no warning here
+
+def check_warnings(model_id, X, Y, warnNumber, training_data):
+    buffer = StringIO() # redirect output
+    sys.stderr=buffer
+    model = H2OGeneralizedLinearEstimator(family="Gaussian", lambda_search=True, Lambda=[0.01])
+    model.train(x=X, y=Y, training_frame=training_data)
+    warn_phrase = "UserWarning"
+    warn_string_of_interest = "slash (/) found"
+    sys.stderr=sys.__stderr__   # redirect printout back to normal path
```

Comment:
should be in finally

Reply:
@honza:

This code is trying to catch the warning phrase.  Therefore, it is not in finally.